modules/nft/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+				for _ in 0..quantity {
+					orml_nft::Module::<T>::mint(&to, class_id, metadata.clone(), data.clone())?;
+				}
+
+				Self::deposit_event(RawEvent::MintedToken(who, to, class_id, quantity));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn transfer(origin, to: T::AccountId, token: (<T as orml_nft::Trait>::ClassId, <T as orml_nft::Trait>::TokenId)) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(token.0).ok_or(Error::<T>::ClassIdNotFound)?;
+				let data: T::ConvertClassData = From::from(class_info.data);
+				ensure!(data.into().properties.contains(&ClassProperty::Transferable), Error::<T>::NonTransferable);
+
+				let token_info = orml_nft::Module::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenIdNotFound)?;
+				ensure!(who == token_info.owner, Error::<T>::NoPermission);
+
+				orml_nft::Module::<T>::transfer(&who, &to, token)?;
+
+				Self::deposit_event(RawEvent::TransferedToken(who, to, token.0, token.1));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn burn(origin, token: (<T as orml_nft::Trait>::ClassId, <T as orml_nft::Trait>::TokenId)) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(token.0).ok_or(Error::<T>::ClassIdNotFound)?;
+				let data: T::ConvertClassData = From::from(class_info.data);
+				ensure!(data.into().properties.contains(&ClassProperty::Burnable), Error::<T>::NonBurnable);
+
+				let token_info = orml_nft::Module::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenIdNotFound)?;
+				ensure!(who == token_info.owner, Error::<T>::NoPermission);
+
+				orml_nft::Module::<T>::burn(&who, token)?;
+				let owner: T::AccountId = T::ModuleId::get().into_sub_account(token.0);
+				let data: T::ConvertTokenData = From::from(token_info.data);
+				let data: TokenData = data.into();
+				<T as Trait>::Currency::unreserve(&owner, data.deposit);
+				<T as Trait>::Currency::transfer(&owner, &who, data.deposit)?;
+
+				Self::deposit_event(RawEvent::BurnedToken(who, token.0, token.1));
+				Ok(())
+			})?;
+		}
+
+		#[weight = 10_000]
+		pub fn destroy_class(origin, class_id: <T as orml_nft::Trait>::ClassId, dest: T::AccountId) {
+			with_transaction_result(|| {
+				let who = ensure_signed(origin)?;
+				let class_info = orml_nft::Module::<T>::classes(class_id).ok_or(Error::<T>::ClassIdNotFound)?;
+				ensure!(who == class_info.owner, Error::<T>::NoPermission);
+				ensure!(class_info.total_issuance == Zero::zero(), Error::<T>::CannotDestroyClass);
+
+				let owner: T::AccountId = T::ModuleId::get().into_sub_account(class_id);
+				let data: T::ConvertClassData = From::from(class_info.data);
+				let data: ClassData = data.into();
+				// reserve balance should equal CreateClassDeposit.
+				ensure!(data.deposit == <T as Trait>::Currency::reserved_balance(&owner), Error::<T>::CannotDestroyClass);
```

Comment:
I don't think we want this check. owner may have reserved balances from other pallets.